Networking/Networking/Model/Product/Product.swift

```diff
@@ -266,17 +266,18 @@ public struct Product: Codable {
 
         let regularPrice = try container.decodeIfPresent(String.self, forKey: .regularPrice)
 
+        let onSale = try container.decode(Bool.self, forKey: .onSale)
+
         // Even though a plain install of WooCommerce Core provides string values,
         // some plugins alter the field value from String to Int or Decimal.
         var salePrice = ""
         if let parsedSalePriceString = container.failsafeDecodeIfPresent(stringForKey: .salePrice) {
-            salePrice = parsedSalePriceString
+            salePrice = (onSale && parsedSalePriceString.isEmpty) ? "0" : parsedSalePriceString
         } else if let parsedSalePriceDecimal = container.failsafeDecodeIfPresent(decimalForKey: .salePrice) {
-            salePrice = NSDecimalNumber(decimal: parsedSalePriceDecimal).stringValue
+            let stringPrice = NSDecimalNumber(decimal: parsedSalePriceDecimal).stringValue
+            salePrice = (onSale && stringPrice.isEmpty) ? "0" : stringPrice
```

Comment:
It doesn't seem to be necessary to check for `stringPrice.isEmpty` in here. It looks like that will always be `false`. The `parsedSalePriceDecimal` is non-`nil` so it will always have a value so any `stringValue` conversion should succeed. 🤔